lib/ClangImporter/ImportMacro.cpp

```diff
@@ -424,118 +440,158 @@ static ValueDecl *importMacro(ClangImporter::Implementation &impl,
+    if (tokenI[0].is(clang::tok::numeric_constant)) {
+      if (auto literal = parseNumericLiteral<clang::IntegerLiteral>(impl,
+                                                                 tokenI[0])) {
+        firstValue = llvm::APSInt { literal->getValue(),
+                              literal->getType()->isUnsignedIntegerType() };
+        firstType  = impl.importType(literal->getType(),
+                                     ImportTypeKind::Value,
+                                     isInSystemModule(DC),
+                                     /*isFullyBridgeable*/false);
+      }
+    }
+    // Look through identifiers which are imported as integer constants.
+    else if (tokenI[0].is(clang::tok::identifier) &&
+             tokenI[0].getIdentifierInfo()->hasMacroDefinition()) {
+
+      auto rawID     = tokenI[0].getIdentifierInfo();
+      auto macroInfo = impl.getClangPreprocessor().getMacroInfo(rawID);
+      auto importedID= impl.getNameImporter().importMacroName(rawID, macroInfo);
+      impl.importMacro(importedID, macroInfo);
+
+      if (auto intValue = impl.getIntegerValueForMacro(macroInfo)) {
+        firstValue = intValue->first;
+        firstType  = intValue->second;
+      }
+    }
+    // Couldn't parse an integer for the first value.
+    if (!firstType) { return nullptr; }
+
+    llvm::APSInt secondValue;
+    Type secondType = nullptr;
+
+    // Parse integer literal.
+    if (tokenI[2].is(clang::tok::numeric_constant)) {
+      if (auto literal = parseNumericLiteral<clang::IntegerLiteral>(impl,
+                                                                  tokenI[2])) {
+        secondValue = llvm::APSInt { literal->getValue(),
+                               literal->getType()->isUnsignedIntegerType() };
+        secondType  = impl.importType(literal->getType(),
+                                      ImportTypeKind::Value,
+                                      isInSystemModule(DC),
+                                      /*isFullyBridgeable*/false);
+      }
+    }
+    // Look through identifiers which are imported as integer constants.
+    else if (tokenI[2].is(clang::tok::identifier) &&
+             tokenI[2].getIdentifierInfo()->hasMacroDefinition()) {
+
+      auto rawID     = tokenI[2].getIdentifierInfo();
+      auto macroInfo = impl.getClangPreprocessor().getMacroInfo(rawID);
+      auto importedID= impl.getNameImporter().importMacroName(rawID, macroInfo);
+      impl.importMacro(importedID, macroInfo);
+
+      if (auto intValue = impl.getIntegerValueForMacro(macroInfo)) {
+        secondValue = intValue->first;
+        secondType  = intValue->second;
+      }
+    }
+    // Couldn't parse an integer for the second value.
+    if (!secondType) { return nullptr; }
+
+    // Addition:
+    // (INT1 + INT2)
+    if (tokenI[1].is(clang::tok::plus)) {
+      llvm::APSInt value = firstValue + secondValue;
+      return createMacroConstant(impl, macro, name, DC, firstType,
```

Comment:
Also, all the `createMacroConstant` calls are the same once we have a valid type and APSInt value. Maybe that can be factored out too?